test/test_array.cpp

```diff
@@ -1814,53 +1814,69 @@ TEST(UpperBoundCorrectness)
 
 TEST(ParallelSearchEqualMatch)
 {
-    uint64_t buff[2] = {0, 0};
-    constexpr size_t width = 1;
-    constexpr size_t size = 128;
-    constexpr int64_t key = -1;
-    BfIterator it(buff, 0, width, width, 0);
-    for (size_t i = 0; i < size; ++i) {
-        it.set_value(1); // this is equivalent to set it to -1
-        ++it;
-    }
-    it.move(0);
-    for (size_t i = 0; i < size; ++i) {
-        auto v = sign_extend_value(width, *it);
-        CHECK_EQUAL(v, -1);
-        ++it;
-    }
-    const auto mask = 1ULL << (width - 1);
-    const auto msb = populate(width, mask);
-    const auto search_vector = populate(width, key);
-
-    static auto vector_compare_eq = [](auto msb, auto a, auto b) {
-        return find_all_fields_EQ(msb, a, b);
-    };
-
-    size_t start = 0;
-    const auto end = size;
-    std::vector<size_t> parallel_result;
-    while (start < end) {
-        start = parallel_subword_find(vector_compare_eq, buff, size_t{0}, width, msb, search_vector, start, end);
-        if (start != end)
-            parallel_result.push_back(start);
-        start += 1;
-    }
+    std::mt19937_64 gen64;
+    constexpr size_t buflen = 4;
+    uint64_t buff[buflen];
+    std::vector<int64_t> values;
+    for (size_t width = 1; width <= 64; width++) {
```

Comment:
This loop causes issues with usan: 
`[2024/05/13 13:53:59.835] 1: /data/mci/ee729c6c5743ba288e0d2eb0f20e5111/realm-core/test/test_array.cpp:1823:41: runtime error: shift exponent 64 is too large for 64-bit type 'unsigned long long'`...
In practise, I don't think width can be 64, since we use 1 bit for the sign and the rest bits for the width (max 63). @finnschiermer

Reply:
width *can* be 64, so I'll fix it.